maitake/src/wait/cell.rs

```diff
@@ -88,8 +121,13 @@ impl WaitCell {
             }
         });
 
+        if let Some(prev_waker) = prev_waker {
```

Comment:
Before we would only wake the "old" waker in the cell in the case of some kind of spurious close event.

Now, if you replace one waker with another, the old one gets woken as if the wait completed. This could lead to two wakers fighting over the cell, but hey, whatcha gunna do, don't use a WaitCell.